src/client/app/actions/admin.ts

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import axios from 'axios';
+import { changeBarStacking, changeChartToRender } from './graph';
+import { showErrorNotification, showSuccessNotification } from '../utils/notifications';
+import { getToken } from '../utils/token';
+import { ChartTypes } from '../types/redux/graph';
+import { PreferenceRequestItem } from '../types/items';
+import * as t from '../types/redux/admin';
+import { ActionType, Dispatch, GetState, Thunk } from '../types/redux/actions';
+import { State } from '../types/redux/state';
+
+export function updateDisplayTitle(displayTitle: string): t.UpdateDisplayTitleAction {
+	return { type: ActionType.UpdateDisplayTitle, displayTitle };
+}
+
+export function updateDefaultChartToRender(defaultChartToRender: ChartTypes): t.UpdateDefaultChartToRenderAction {
+	return { type: ActionType.UpdateDefaultChartToRender, defaultChartToRender };
+}
+
+export function toggleDefaultBarStacking(): t.ToggleDefaultBarStackingAction {
+	return { type: ActionType.ToggleDefaultBarStacking };
+}
+
+function requestPreferences(): t.RequestPreferencesAction {
+	return { type: ActionType.RequestPreferences };
+}
+
+function receivePreferences(data: PreferenceRequestItem): t.ReceivePreferencesAction {
+	return { type: ActionType.ReceivePreferences, data };
+}
+
+function markPreferencesNotSubmitted(): t.MarkPreferencesNotSubmittedAction {
+	return { type: ActionType.MarkPreferencesNotSubmitted };
+}
+
+function markPreferencesSubmitted(): t.MarkPreferencesSubmittedAction {
+	return { type: ActionType.MarkPreferencesSubmitted };
+}
+
+
+function fetchPreferences(): Thunk {
+	return (dispatch: Dispatch) => {
+		dispatch(requestPreferences());
+		return axios.get('/api/preferences')
+			.then(response => {
+				dispatch(receivePreferences(response.data));
+				dispatch((dispatch2: Dispatch, getState: GetState) => {
+					const state = getState();
+					dispatch2(changeChartToRender(state.admin.defaultChartToRender));
```

Comment:
I presume we'll be calling `dispatch2` here with all the preferences, right? Is there a way to factor this out more? If not that's fine, just wondering.

Reply:
I would love if you could come up with a way to factor this out, but I think it would require some stuff outside the scope of this PR.

Right now, to add an additional preference, there is a host of things one must change. It would be great to simplify it.